server/src/controllers/MarkerController.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response } from 'express';
+import Marker from '../models/Marker';
+
+// GET endpoint to retrieve all markers with longitude and latitude
+export const getMarkers = async (req: Request, res: Response) => {
+    try {
+        // Retrieve all markers from the database with projection to include only longitude and latitude
+        const markers = await Marker.find({}, 'longitude latitude');
+        return res.status(200).json(markers); // Return retrieved markers as JSON response
+    } catch (err) {
+        console.error('Could not get markers: ', err);
+        return res.status(500).json({ message: 'Internal server error' }); // Handle errors gracefully
+    }
```

Comment:
I think you should make the error messages for **all** the endpoints more descriptive. For someone who is consuming the API, it might not be clear why this request is throwing a `"500 Internal server error"`. 

You already add descriptive messages for successful requests in `saveMarker` and `deleteMarker`; might as well do the same for the failed requests.

You could either add the `console.error` message to the message field:
```suggestion
    } catch (err) {
        const message: string = 'Could not get markers';
        console.error(message, err);
        return res.status(500).json({ message }); // Handle errors gracefully
    }
```

Or just report the `err` itself
```suggestion
    } catch (err) {
        console.error('Could not get markers', err);
        return res.status(500).json({ message: err }); // Handle errors gracefully
    }
```

Reply:
Resolved. Thank you for your suggestion, and I believe that now the API consumer will now be able to understand what potentially went wrong.